src/components/ingestion/FileIngestionEngine.js

```diff
@@ -0,0 +1,33 @@
+import fs from 'fs'
+import AbstractIngestionEngine from '%src/components/ingestion/AbstractIngestionEngine'
+
+/**
+ * The FileIngestionEngine handles processing a video file. It is a concrete
+ * implementation of AbstractIngestionEngine, which reads in a file,
+ * converts it into an MPEG-TS stream represented as a series of Payloads,
+ * and then pipes that to Kafka.
+ *
+ * @param  {string}  path The path of the file to be ingested
+ */
+class FileIngestionEngine extends AbstractIngestionEngine {
+	constructor(path) {
+		if (!path) {
+			throw new Error('path required') // TODO: should probably be FileInstantiationError
```

Comment:
I think Dan covers this already (let's either use the custom error type or remove the TODO, preference for the former), but either way the error message should be more descriptive.

Reply:
“Path absolutely required?”

Reply:
I was thinking more like "FileIngestionEngine must be instantiated with a file path"

Reply:
Done.